36954 tags docs (#6622)
camunda/camunda-docs: docs/components/concepts/process-instance-creation.md

A process can also have one or more [timer start events](/components/modeler/bpmn/timer-events/timer-events.md#timer-start-events). An instance of the process is created when the associated timer is triggered. Timers can also trigger periodically.

## Tags (8.8+)

> Process instance tags are lightweight, immutable labels you can attach when creating a process instance via the v2 API or SDK. They help downstream workers and external systems make quick routing or decision choices without inspecting full variable payloads.

Reviewer: 🔧 I think we don't call it "v2 API" in the docs 😅 We could add a link to the endpoint?

- Tags are exported with the process instance and with job entities starting in 8.8 by the default exporters.
- Tags are not shown in web applications for now (e.g., Operate, Tasklist) in 8.8 — they are API/SDK-only metadata.

### Use cases

- Lightweight correlation (e.g., `businessKey:1234`).
- Routing or selective processing (`priority:high`, `region:emea`).
- Downstream external lookups (e.g. `crmId:123`).

Do not place sensitive or personal data (PII) in tags: they are broadly propagated with jobs and exports!

### Examples

Create with tags:

```
curl -L 'http://localhost:8080/v2/process-instances' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{
  "processDefinitionId": "order-process",
  "processDefinitionVersion": 3,
  "tags": ["priority:high","business_key:1234","region:emea"],
  "variables": { "orderId": "1234" }
}'
```

### Best practices

- Prefer a `key:value` or `key` style to keep tags self-describing.
- Keep values short; use process variables for larger or mutable data.
- Use consistent prefixes for organizational scoping (e.g., `dept:`, `env:`).
- Use variables (not tags) for mutable or large data.

> 

Author: > link to the tags definition
can you elaborate where this should be? glossary?

A process can also have one or more [timer start events](/components/modeler/bpmn/timer-events/timer-events.md#timer-start-events). An instance of the process is created when the associated timer is triggered. Timers can also trigger periodically.

> ## Tags (8.8+)

Reviewer 2: ```suggestion
## Tags
```

> Process instance tags are lightweight, immutable labels you can attach when creating a process instance via the API or SDK. They help downstream workers and external systems make quick routing or decision choices without inspecting full variable payloads.

Reviewer 2: ```suggestion
Process instance tags are lightweight, immutable labels you can attach when creating a process instance via the API or Clients. Tags get inherited by all jobs created from that instance. They help downstream workers and external systems make quick routing or decision choices without inspecting full variable payloads.
```

> ### Definition

Reviewer 2: ```suggestion
### Tag format and constraints
```

- A tag is a case-sensitive string.
- Format (regex): `^[A-Za-z][A-Za-z0-9_\-:.]{0,99}$`
    - Must start with a letter (A-Z or a-z).
    - Remaining characters may be alphanumeric, underscore (`_`), hyphen (`-`), colon (`:`), or dot (`.`).
- Length: 1–100 characters.
- Maximum 10 unique tags per process instance (duplicates are ignored).

> - Order is not guaranteed; treat the set as unordered.

Reviewer 2: ```suggestion
- Order is not guaranteed; treat the set as unordered.
- Tags cannot be modified after creation
```

Validation failures during process instance creation (too many tags, invalid pattern/lengt) cause the create request to be rejected with a 4xx error.

### Semantics

- Tags are included in process instance search responses and on activated job payloads.

> - Tags are immutable after creation for now (cannot be added, changed, or removed later).

Reviewer 2: ```suggestion
- Tags are immutable after creation - cannot be added, changed, or removed after process instance has been created.
```

- Search filtering uses AND semantics: an instance must contain all requested tags (it may contain additional tags). Partial or wildcard matching is not supported for now.

> - Tags are exported with the process instance and with job entities starting in 8.8 by the default exporters.

Reviewer 2: ```suggestion
- Tags are supported for process instances and with job entities started in 8.8.
```

> - Tags are not shown in web applications for now (e.g., Operate, Tasklist) in 8.8 — they are API/SDK-only metadata.

Reviewer 2: ```suggestion
- Tags are not shown in web applications (e.g., Operate, Tasklist) — they are API/Client-only metadata.
```

### Use cases

- Routing and prioritization (e.g. `priority:high`).

> - Business / domain identifiers from internal or third‑party systems (e.g. `businessKey:1234`, `customerId:7890`, `orderId:4567`).

Reviewer 2: ```suggestion
- Business / domain identifiers from internal or third‑party systems (e.g. `team:accounting`, `started:manually`).
```

- Cross-system correlation keys without exposing full variable payloads. (e.g. `traceId:abcd-1234`, `crmId:3004`).
- Analytics segmentation (e.g. `region:emea`, `channel:web`).
- Feature rollout / experiment grouping (e.g. `experiment:checkout-v2`).

> - Environment or tenant-like labeling where full multi-tenancy isn’t required (e.g. `env:staging`).

Reviewer 2: ```suggestion
```

### Guidelines

> - Do not store secrets or PII; tags propagate with jobs and exports.

Reviewer 2: ```suggestion
- Do not store secrets or PII; tags are propagated with jobs and exporters.
```

- Prefer concise `key:value` or `key` patterns for consistency.
- Use variables (not tags) for mutable or large data.
- Establish internal naming conventions (for example, prefixes like `env:`, `dept:`) for governance. 

### Examples

Create with tags:

```
curl -L 'http://localhost:8080/v2/process-instances' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{
  "processDefinitionId": "order-process",
  "processDefinitionVersion": 3,

>   "tags": ["priority:high","business_key:1234","region:emea"],

Reviewer 2: ```suggestion
  "tags": ["channel:web","region:emea"],
```
